File: SWEA/D3/1206. ［S／W 문제해결 기본］ 1일차 － View/［S／W 문제해결 기본］ 1일차 － View.py
for k in range(10):
    a=int(input())
    arr=list(map(int,input().split()))
    count=0
    for i in range(2, a-2):
        M=max(arr[i-2], arr[i-1], arr[i+1],arr[i+2])
        if arr[i]>M:
            count+=arr[i]-M
    print(f'#{k+1} {count}')

File: SWEA/D3/1206. ［S／W 문제해결 기본］ 1일차 － View/［S／W 문제해결 기본］ 1일차 － View.java
import java.util.Scanner;

public class Solution {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        for (int tc = 1; tc <= 10; tc++) {
            int T = sc.nextInt();
            int[] arr = new int[T];

            for (int j = 0; j < T; j++) {
                arr[j] = sc.nextInt();
            }

            int result = 0;

            for (int p = 2; p < T - 2; p++) {
                int maxL = Math.max(arr[p - 2], arr[p - 1]);
                int maxR = Math.max(arr[p + 1], arr[p + 2]);
                int M = Math.max(maxL, maxR);

                if (arr[p] > M) {
                    result += (arr[p] - M);
                }
            }

            System.out.printf("#%d %d\n", tc, result);
        }
    }
}
